fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/eventslifecycle/MixinEntityType.java

```diff
@@ -0,0 +1,42 @@
+package net.fabricmc.fabric.mixin.eventslifecycle;
+
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.fabricmc.fabric.api.event.entity.EntityTickCallback;
+import net.fabricmc.fabric.impl.event.EntityTypeCaller;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.profiler.Profiler;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(EntityType.class)
+public class MixinEntityType<T extends Entity> implements EntityTypeCaller {
+	private EntityTickCallback event;
+
+	@Override
+	public EntityTickCallback getEntityEvent() {
+		if (event == null) {
```

Comment:
Is there any actual reason to lazily load this ? I would say, either create the event only when a mod wants to register a callback, or create it at initialization and make the field `final`.